docs/hibernating-clusters.md

```diff
@@ -0,0 +1,146 @@
+# Hibernating Clusters
+
+## Overview
+
+Starting with OpenShift version 4.4.8, clusters can be stopped and started by simply shutting
+down their machines and starting them back up. The only additional requirement to bring a cluster
+back up from a stopped state is that CSRs be approved for the cluster's nodes when certificates
+have expired since the cluster was active. This typically happens if the cluster is hibernated
+within the first 24 hours before initial cert rotation takes place. After this phase, certs will
+be valid for 30 days.
+
+Hive can automate the process of stopping/starting clusters via its API by allowing the user to
+set a the desired state of the cluster in the ClusterDeployment spec. Both API and controller
+changes are required to support this feature.
+
+## API Changes
+
+The ClusterDeploymentSpec should allow setting whether machines are in a running state or in
+a hibernating state.
+
+```go
+type ClusterRunningState string
+
+const (
+  // RunningClusterRunningState is the default state of a cluster after it has
+  // been installed. All of its machines should be running.
+  RunningClusterRunningState string = "Running"
+
+  // HibernatingClusterRunningState is used to stop the machines belonging to a cluster
+  // and move it to a running state.
+  HibernatingClusterRunningState string = "Hibernating"
```

Comment:
Can we name var `ClusterRunningState` as `ClusterState` and status strings as `ClusterRunningState` & `ClusterHibernatingState`?